src/projects/apps/auth/authentication.md

```diff
@@ -0,0 +1,54 @@
+
+### Username & Password
+
+Username & Password authentication is the most basic form of authentication offered in an App, it will allow users to log into an App by a unique username and password that is stored in the platform.
+The authentication flow is all handled within the platform as is the list of App users and their information.
+Once a user is created, they can be assigned to one or more Groups which will authorize them to access resources and pages within the App.
+
+### OAuth2/OpenID
+
+The OAuth2 and OpenID frameworks are very similar to each other and both allow you to utilise another authentication provider (e.g. Microsoft or Google) for your Apps authentication. If you have ever used an app or website that gives you the option to log in with Google, LinkedIn or Microsoft then this is OAuth2 and OpenID behind the scenes.
+In Apps, you can use any :docs-link[Identity Provider]{id="admin/identity_providers"} that you have set up in your platform as the authentication provider for your App.
+
+The main difference between OAuth2 and OpenID in Apps is that OpenID guarantees certain information can be retrieved for the user that has logged in such as username, name, date of birth, address, email and more.
+This is why when your App only uses OAuth2 there is no username in your App Users table as the platform cannot guarantee that it is provided that information.
+
+In both OAuth2 and OpenID authenticated apps you need to provide the ProfileURL (can also be known as the User Info endpoint) as this is used to fetch the user ID from the authentication provider you are using.
+For example for Google the profile URL is [https://www.googleapis.com/oauth2/v1/userinfo](https://www.googleapis.com/oauth2/v1/userinfo) and for Microsoft it is [https://graph.microsoft.com/oidc/userinfo](https://graph.microsoft.com/oidc/userinfo).
+
+You also need to provide the path to the ID field so if your reponse from the Profile API looks like this:
+```json
+{
+    "sub": "OLu859SGc2Sr9ZsqbkG-QbeLgJlb41KcdiPoLYNpSFA",
+    "name": "Mikah Ollenburg",
+    "family_name": "Ollenburg",
+    "given_name": "Mikah",
+    "picture": "https://graph.microsoft.com/v1.0/me/photo/$value",
+    "email": "mikoll@contoso.com"
+}
+```
+then the path to the ID field will be `sub`.
+
+You also need to provide the relevant scopes to access some of the user information. Typically you require the `profile` scope to get any name information and you require the `email` scope to get information about the users email.
+
+If you are using OpenID as your authentication protocol then you'll need to also provide a path to the email field (this would be `email` from the above JSON) and a path to the name field (this could be either the `name` or `given_name` field in the above JSON).
+When using OpenID it is advised to use at a minimum the following scopes `openid profile email`.
```

Comment:
openid scope is required for it to be opened but additional scopes are able to be added

Reply:
What do you mean by "opened"?

Reply:
sorry miss typed. Without the `openid` scope it's not considered to confirm to the opened protocol

Reply:
Ah, thanks, will add that